hardware/1Wire/1WireByKernel.cpp

```diff
@@ -551,7 +543,8 @@ void C1WireByKernel::GetDevice(const std::string& deviceName, /*out*/_t1WireDevi
 	device.family = ToFamily(deviceName.substr(0, 2));
 
 	// Device Id (6 chars after '.')
-	std_to_upper(deviceName.substr(3, 3 + 6 * 2), device.devid);
+	const auto str = deviceName.substr(3, 3 + 6 * 2);
+	std::transform(str.begin(), str.end(), device.devid.begin(), ::toupper);
```

Comment:
We do have a 'stdupper' upper function in the helper.cpp/h file

Reply:
That's not going to work. Source and destination are not the same.

Reply:
Eh kind of did it. Requires one more line :).